packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/Legend/TooltipConfig.tsx

```diff
@@ -93,9 +96,37 @@ export const TooltipConfig: FC<Props> = ({ fields }) => {
         visualizationConfig.chartConfig,
     ]);
 
+    const handleEditorOnChange = (value: string | undefined) => {
+        setTooltipValue(value ?? '');
+    };
+
     const [show, setShow] = useState<boolean>(
         isCartesianChart ? !!visualizationConfig.chartConfig.tooltip : false,
     );
+    const [monacoOptions, setMonacoOptions] = useState<
+        EditorProps['options'] | undefined
+    >();
+
+    useEffect(() => {
+        /** Creates a container that belongs to body, outside of the sidebar
+         * so we can place the autocomplete tooltip and it doesn't overflow
+         * CSS for this component is set on `monaco.css`
+         */
+        const containerId = 'monaco-overflow-container';
+        let container = document.getElementById(containerId);
+        if (!container) {
+            const wrapper = document.createElement('div');
+            wrapper.className = 'monaco-editor';
+            container = document.createElement('div');
+            container.id = containerId;
+            wrapper.appendChild(container);
+            document.getElementById('root')?.appendChild(wrapper);
+        }
+        setMonacoOptions({
+            ...MONACO_DEFAULT_OPTIONS,
+            overflowWidgetsDomNode: container,
+        });
+    }, [monacoOptions]);
```

Comment:
this causes an infinite render, can you replace `useEffect` with `useDeepCompareEffect` from `react-use`?